course/templates/parsons_question.html

```diff
@@ -1,47 +1,103 @@
                 </div>
+                {% endfor %}
             </div>
-            <div class="col-md-6">
-                <h5>Your Solution</h5>
-                <div class="right-container">
+        </div>
+        <div class="col-md-6">
+            <h5>Your Solution</h5>
+            <div class="right-container">
 
-                </div>
             </div>
         </div>
     </div>
-    <script>
-        drake = dragula([$(".left-container")[0], $('.right-container')[0]])
-            .on('drop', function (e1, target, source, sibling) {
-                calculate_source_code();
-            })
+</div>
+<script>
+    drake = dragula([$(".left-container")[0], $(".right-container")[0]])
+        .on("drop", function (e1, target, source, sibling) {
+            determine_indents();
+            remove_indents();
+            calculate_source_code();
+        });
 
-        function calculate_source_code() {
-            let code = $('.right-container').children().text();
-            $('#source_code').val(code);
-        }
-    </script>
-    <div class="my-3">
-        <form method="post" enctype="multipart/form-data">
-            {% csrf_token %}
+    function calculate_source_code() {
+        let parsons_lines = $(".right-container pre.code_lines");
+        let code = "";
+        parsons_lines.each(function (i, e) {
+            let line_value = $(e);
+            let line = line_value.text();
+            code += line + "\n";
+        });
+        $("#source_code").val(code);
+    }
 
-            <input type="hidden" id="source_code" name="code" class="form-control">
+    function determine_indents() {
+        let parsons_lines = $(".right-container pre.code_lines");
+        let count = 0;
+        parsons_lines.each(function (i, e) {
+            let line_value = $(e);
+            let line = line_value.text().trim();
+            line_value.text(line);
+            if (line.charAt(line.length - 1) == "{") {
+                line_value.text(indent(line, count));
+                count++;
+            }
+            else if (line.charAt(line.length - 1) == "}") {
+                count--;
+                line_value.text(indent(line, count));
+            }
+            else if (count > 0) {
+                line_value.text(indent(line, count));
+            }
+        });
+    }
 
-            <div class="form-group my-3">
-                <button type="submit" class="btn btn-success"> Submit</button>
-            </div>
-        </form>
-    </div>
+    function remove_indents() {
+        let parsons_lines = $(".left-container pre.code_lines");
+        parsons_lines.each(function (i, e) {
+            let line_value = $(e);
+            let line = line_value.text().trim();
+            line_value.text(line);
+        });
+    }
+
+    function indent(str, numOfIndents, opt_spacesPerIndent) {
+        str = str.replace(/^(?=.)/gm, new Array(numOfIndents + 1).join("\t"));
```

Comment:
I have no idea what this function does and can't figure it out from the code.
maybe add some comments.

Reply:
Okay sounds good